pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -242,52 +239,63 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		appliedManifestWorkClient: n.appliedManifestWorkClient,
 	}
 
-	// If there are some invalid feature gates of registration, will output condition `ValidRegistrationFeatureGates` False in Klusterlet.
-	if klusterlet.Spec.RegistrationConfiguration != nil && len(klusterlet.Spec.RegistrationConfiguration.FeatureGates) > 0 {
-		featureGateArgs, invalidFeatureGates := helpers.FeatureGatesArgs(
-			klusterlet.Spec.RegistrationConfiguration.FeatureGates, helpers.ComponentSpokeRegistrationKey)
-		if len(invalidFeatureGates) == 0 {
-			config.RegistrationFeatureGates = featureGateArgs
-			// TODO: after the 0.10.0 is released, change back to UpdateKlusterletConditionFn
-			_, _, _ = helpers.UpdateKlusterletStatus(ctx, n.klusterletClient, klusterletName, helpers.ReplaceKlusterletConditionFn(
-				spokeRegistrationFeatureGatesInvalid,
-				metav1.Condition{
-					Type: spokeRegistrationFeatureGatesValid, Status: metav1.ConditionTrue, Reason: "FeatureGatesAllValid",
-					Message: "Registration feature gates of klusterlet are all valid",
-				}))
-		} else {
-			invalidFGErr := fmt.Errorf("there are some invalid feature gates of registration: %v ", invalidFeatureGates)
-			// TODO: after the 0.10.0 is released, change back to UpdateKlusterletConditionFn
-			_, _, _ = helpers.UpdateKlusterletStatus(ctx, n.klusterletClient, klusterletName, helpers.ReplaceKlusterletConditionFn(
-				spokeRegistrationFeatureGatesInvalid,
-				metav1.Condition{
-					Type: spokeRegistrationFeatureGatesValid, Status: metav1.ConditionFalse, Reason: "InvalidFeatureGatesExisting",
-					Message: invalidFGErr.Error(),
-				}))
-			return invalidFGErr
+	// Do not use "conditions := &klusterlet.Status.Conditions", otherwise will override conditions
+	// set by other controllers
+	conditions := &[]metav1.Condition{}
```

Comment:
why?

Reply:
the `conditions` we set here likely mean "additional" conditions.
For example, if we use `conditions := &klusterlet.Status.Conditions`, at the beginning there is a condition `WorkDesiredDegraded=true` in the `klusterlet.Status.Conditions`, and another controller updates a condition `WorkDesiredDegraded=falst` **at the same time**, and at the end of this `sync` we update the status using `UpdateKlusterletStatus` which will update WorkDesiredDegraded condition back to `true`.

Reply:
we should not have two controllers update the same condition type.

Reply:
yes, that is why we do not use `conditions := &klusterlet.Status.Conditions` here,  we use `conditions := &[]metav1.Condition{}` to set conditions that are only generated by the current controller.